src/jit/flowgraph.cpp

```diff
@@ -5746,14 +5746,36 @@ void          Compiler::fgFindBasicBlocks()
 
     if (compIsForInlining())
     {
+        bool hasReturnBlocks = false;
+        bool hasMoreThanOneReturnBlock = false;
+
+        for (BasicBlock* block = fgFirstBB; block; block = block->bbNext)
+        {
+            if (block->bbJumpKind == BBJ_RETURN)
+            {
+                if (hasReturnBlocks)
+                {
+                    hasMoreThanOneReturnBlock = true;
+                    break;
+                }
+
+                hasReturnBlocks = true;
+            }
+        }
+
+        if (!hasReturnBlocks)
+        {
+            impInlineInfo->inlineResult->NoteFatal(InlineObservation::CALLEE_DOES_NOT_RETURN);
+            return;
+        }
+
```

Comment:
Yes, exceptions are slow. But the point is that `X is postdominated by a throw' does not imply `X is not performance sensitive`.

This is a tricky thing to get right because there are cases where a relatively large amount of computation is done solely for the purpose of throwing a more detailed exception.

My preference would be to start off conservatively here, only matching simple methods with this heuristic, for instance methods that are loop-free and relatively small. This should catch most of the idiomatic throw helper cases.